### PR TITLE
Pgoapi purge

### PR DESCRIPTION
**Pull request by invisiblek**
Kill our local copy of pgoapi and instead pull in a fresh copy from tejado's repo. Fixups along the way to handle the api changes.

### CODE REVIEW

**Comment by medghaim** [comment]
I know my +1 doesn't count, but thorough testing reveals all works fine and dandy. _+1_

**Comment by invisiblek** [comment]
"medghaim commented 2 minutes from now"

wizard!

**Comment by invisiblek** [comment]
People are going to need to re-run requirements.txt as well as probably purge their db (unless we want to bother with writing migration scripts)

**Comment by medghaim** [comment]
Ew no.

(is this from the future too?)

**Comment by invisiblek** [comment]
"medghaim commented 2 minutes from now"

time traveler

**Comment by Chlodochar** [comment]
Related #671

**Comment by medghaim** [comment]
Yeah, invisiblek built off your earlier PR. Your commits are still up there somewhere!

**Comment by chuyskywalker** [comment]
`pgoapi INFO` level logging used to be silenced, I think.

Modify `runserver.py` with

```
    logging.getLogger('pgoapi.pgoapi').setLevel(logging.WARNING)
    logging.getLogger('pgoapi.rpc_api').setLevel(logging.INFO)
```

**Comment by Chlodochar** [comment]
Nice. I'm not able to test this soon, but would be very happy when this is merged! Should @tejado do a release of this commit?

**Comment by FrostTheFox** [comment]
Definitely runs. Going to compare pokemon amounts here in a few.

**Comment by chuyskywalker** [comment]
LGTM so far so good. I'll let it churn overnight, see if anything weird pops

**Comment by medghaim** [comment]
Always a good sign when things run, @FrostTheFox :D

**Comment by FrostTheFox** [comment]
I've ran both latest dev and this at the exact same time, and allowed them to finish 1 full hex, before killing them and screenshotting both. Different Google account for each one. They're both running -sd 5 and -st 7.

The left side is latest dev. Right is pgoapi-purge. http://i.imgur.com/PozqW3l.png

Most pokemon appear on both. But pgoapi-purge definitely has a few more than dev.

Everything else ran smoothly, pgoapi-purge a little slower, but nothing big.

+1

**Comment by FrostTheFox** [comment]
Whoops, didn't mean to close

**Comment by medghaim** [comment]
Awesome! Thanks for the comparison. It'll be more prominent in other areas (Tokyo, Japan for example) due to how dev uses hilbert to generate the path. 

For the most part though, hilbert walking didn't miss many pokemon, but it did miss some and needed fixing.

**Comment by medghaim** [comment]
Yeah, pgoapi runs a little slower because we're grabbing an insane amount of cells, I'll play around with that!

**Comment by FrostTheFox** [comment]
@medghaim beginning same test of tokyo

**Comment by Chlodochar** [comment]
I've ran #671 before and it also resulted in more Pokemon with the new pgoapi. Not sure if that was because of the update or something else at the time.

**Comment by macgngsta** [comment]
`python runserver.py -st 5`
- cleared the pogom.db for each test
- ran 1 full cycle on dev = 4dba6e6, pogoapi-purge = 24845a0

1st pic = dev ~ 51 pokemon found
dev_4dba6e6

2nd pic = pgoapi ~ 47 pokemon found
pgoapi_ 24845a0

based on the gif, it does seem to produce a more distributed pokemon set, vs sparse clumped toward the outside
giphy

ill attach the console logs. 
[dev_4dba6e6.txt](https://github.com/AHAAAAAAA/PokemonGo-Map/files/395366/dev_4dba6e6.txt)
[pgoapi_ 24845a0.txt](https://github.com/AHAAAAAAA/PokemonGo-Map/files/395368/pgoapi_.24845a0.txt)

**Comment by chuyskywalker** [comment]
I've got two accounts running and one of them seemed to have gotten stuck not being able to login reliably:

```
2016-08-02 06:45:32,898 [ search_worker_0][        pgoapi][   ERROR] Login failed - unexpected server response!
2016-08-02 06:45:32,898 [ search_worker_0][        search][   ERROR] Failed to login to Pokemon Go. Trying again in 15 seconds
2016-08-02 06:45:47,921 [ search_worker_0][      auth_ptc][    INFO] Login for: usernamegoeshere
2016-08-02 06:45:49,456 [ search_worker_0][      auth_ptc][    INFO] PTC Login successful
2016-08-02 06:45:50,870 [ search_worker_0][        pgoapi][   ERROR] Login failed - unexpected server response!
2016-08-02 06:45:50,872 [ search_worker_0][        search][   ERROR] Failed to login to Pokemon Go. Trying again in 15 seconds
2016-08-02 06:46:05,889 [ search_worker_0][      auth_ptc][    INFO] Login for: usernamegoeshere
2016-08-02 06:46:08,963 [ search_worker_0][      auth_ptc][    INFO] PTC Login successful
2016-08-02 06:46:09,936 [ search_worker_0][        pgoapi][   ERROR] Login failed - unexpected server response!
2016-08-02 06:46:09,936 [ search_worker_0][        search][   ERROR] Failed to login to Pokemon Go. Trying again in 15 seconds
2016-08-02 06:46:24,948 [ search_worker_0][      auth_ptc][    INFO] Login for: usernamegoeshere
2016-08-02 06:46:27,509 [ search_worker_0][      auth_ptc][    INFO] PTC Login successful
2016-08-02 06:46:28,678 [ search_worker_0][        pgoapi][   ERROR] Login failed - unexpected server response!
2016-08-02 06:46:28,678 [ search_worker_0][        search][   ERROR] Failed to login to Pokemon Go. Trying again in 15 seconds
2016-08-02 06:46:43,704 [ search_worker_0][      auth_ptc][    INFO] Login for: usernamegoeshere
2016-08-02 06:46:45,754 [ search_worker_0][      auth_ptc][    INFO] PTC Login successful
2016-08-02 06:46:46,301 [ search_worker_0][        pgoapi][   ERROR] Login failed - unexpected server response!
2016-08-02 06:46:46,301 [ search_worker_0][        search][   ERROR] Failed to login to Pokemon Go. Trying again in 15 seconds
```

It's weird that the `auth_ptc` says it worked, but then the api level gets all weird about it.

Once the thread got like that, it was stuck doing that. Don't have any more debug that that so far. Restarted the server and the account is still working, so it was definitely us getting the PgoAPI instance into a weird state and being stuck there.

**Comment by Dagothur** [comment]
Is it possible to get 100% Pokemon coverage with enough tinkering or will there always be deficiency?

**Comment by invisiblek** [comment]
Weird my hexes are off slightly now (they were perfectly aligned before). I'm using raw lat/long for them too, so they should be lined up.

**Comment by chuyskywalker** [comment]
Sometime during the overnight run I lost one of my `search_worker` threads. I expect it got some uncaught exception and dumped out, but my scrollback buffer is to short to see what happened.

**Comment by invisiblek** [comment]
I have 20 workers going, been running strong all night.

**Comment by mgoodsell91** [comment]
Exception:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/req/req_set.py", line 736, in install
    requirement.uninstall(auto_confirm=True)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/req/req_install.py", line 742, in uninstall
    paths_to_remove.remove(auto_confirm)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/req/req_uninstall.py", line 115, in remove
    renames(path, new_path)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip/utils/__init__.py", line 267, in renames
    shutil.move(old, new)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 303, in move
    os.unlink(src)
OSError: [Errno 13] Permission denied: '/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/google/protobuf/__init__.py'

Getting this as an error when running "pip install -r requirements.txt" 

Results in this error when searching 
Traceback (most recent call last):
  File "runserver.py", line 35, in <module>
    from pogom.search import search_overseer_thread, fake_search_loop
  File "/Users/MattGoodsell/Downloads/PokemonGo-Map-pgoapi-purge/pogom/search.py", line 28, in <module>
    from pgoapi import PGoApi
ImportError: No module named pgoapi
Any ideas?

**Comment by chuyskywalker** [comment]
Here we go:

```

2016-08-02 14:15:44,595 [ search_worker_0][      auth_ptc][    INFO] PTC Login successful
Exception in thread search_worker_0:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/src/app/pogom/search.py", line 203, in search_worker_thread
    check_login(args, account, api, position)
  File "/usr/src/app/pogom/search.py", line 239, in check_login
    while not api.login(account['auth_service'], account['username'], account['password'], position[0], position[1], position[2], False):
  File "/usr/src/app/src/pgoapi/pgoapi/pgoapi.py", line 129, in login
    response = self.get_player()
  File "/usr/src/app/src/pgoapi/pgoapi/pgoapi.py", line 84, in function
    return request.call()
  File "/usr/src/app/src/pgoapi/pgoapi/pgoapi.py", line 182, in call
    response = request.request(self._api_endpoint, self._req_method_list, self.get_position())
  File "/usr/src/app/src/pgoapi/pgoapi/rpc_api.py", line 116, in request
    raise NotLoggedInException()
NotLoggedInException
```

That's causing the search_worker thread to dump out. Needs a bit of exception catching.

**Comment by chuyskywalker** [comment]
Alright, ignore my issue, it's not really a pgoapi upgrade problem but a "I didn't handle thread exceptions well enough" problem. I'll task that in a separate PR

**Comment by chuyskywalker** [comment]
LGTM -- I don't know if we hold off for the beehive misalignement or not though

**Comment by invisiblek** [comment]
nobody else is seeing the misalignment at all? the locations i'm using literally have the exact lat or long and they're showing up misaligned =/

@medghaim any idea here? (see screenshot from above)

**Comment by chuyskywalker** [comment]
I've fixed the reliability issues over in this [a pr against this branch](https://github.com/AHAAAAAAA/PokemonGo-Map/pull/3262/files?w=0) (that's the diff link excluding whitespace since I had to indent a lot for the fix). Should clear up and prevent any weird outliers for getting search threads lost.

**Comment by Talamar1** [comment]
@invisiblek I've been seeing the same alignment type issues the past few days.  I grabbed the latest dev build and the alignment is fixed but now there is massive overlap on the edges
image

**Comment by Crazy-Drew** [comment]
Can anyone give me good clear instructions on how to run more than 1 worker? simultaneously? That would be great!

**Comment by invisiblek** [comment]
This fixes the hexes being off: https://github.com/tejado/pgoapi/pull/171

**Comment by Crazy-Drew** [comment]
@FrostTheFox or @macgngsta Could you do a search with your workers for these coordinates? I am getting very few spawns, want to see if it's my end? I am on last night's build. **44.619585 -65.759572**

**Comment by invisiblek** [comment]
pgoapi updated with the hex fix, latest commit pulls in that change

+111111111111111111

**Comment by DomGrieco** [comment]
@Crazy-Drew really not the place to be posting stuff like that

**Comment by AHAAAAAAA** [comment]
This looks ready. If anyone has any objections, speak now or forever handle issues from here on out.

**Comment by Chlodochar** [comment]
Yay!

**Comment by aldarund** [comment]
@AHAAAAAAA https://github.com/AHAAAAAAA/PokemonGo-Map/issues/3370

**Comment by hack3rjack** [comment]
Also getting error when installing requirements.txt on line 18 for pgoapi, see gist

https://gist.github.com/hack3rjack/4a95b33dd811721d8b91389eeb44590f

**Comment by aldarund** [comment]
@hack3rjack its just a warnings that depends on your python build, all fine with it

**Comment by hack3rjack** [comment]
ok then, thanks @aldarund

File: runserver.py [hunk 1]
```diff
@@ -2,28 +2,30 @@
...
+# This is a leftover directory from the time pgoapi was embedded in PokemonGo-Map
+# The empty directory will cause problems with `import pgoapi` so it needs to go
+oldpgoapiPath = os.path.join(os.path.dirname(__file__), "pogom/pgoapi")
+if os.path.isdir(oldpgoapiPath):
+    log.info("I found %s, but its no longer used. Going to remove it...", oldpgoapiPath)
+    shutil.rmtree(oldpgoapiPath)
+    log.info("Done!")
+
+# Ensure user has updated to the new api
+newpgoapiPath = os.path.join(os.path.dirname(__file__), "src/pgoapi")
```
Comment (Yossi) [inline runserver.py]: this is not where pip installs pgoapi. in fact, there is no src/ directory

Comment (invisiblek) [inline runserver.py]: go ahead and use your super brain to improve it then

Comment (aldarund) [inline runserver.py]: Will he is right, this check dont work and simply stop project from launching

Comment (invisiblek) [inline runserver.py]: Works perfectly fine here https://gist.github.com/invisiblek/f0e5f8b76eec1b23ebf1be3c5283072c

Comment (aldarund) [inline runserver.py]: And? This is really braindead check. You basically rely that pip Will clone imto cuurent due and not clean it. This folder not even user, its just leftoff of installation . And recent pip Will delete this folder.

Comment (aldarund) [inline runserver.py]: *due - directory
*User - used

Comment (Crazy-Drew) [inline runserver.py]: Works awesome over here, definitely seeing way more pokemon now! I am loving this already!

Comment (invisiblek) [inline runserver.py]: Fixed in latest HEAD on develop.

In the future be a bit more respectful of people here. Calling anyone's code braindead is pretty fucking rude.

Comment (Yossi) [inline runserver.py]: I'm sorry. That was in the heat of the moment as the code was crashing right as i needed it.

I'll edit that comment and it wont happen again

**Pull request merged**